reana-ui/src/util.js

```diff
@@ -160,6 +160,13 @@ function parseWorkflowDates(workflow) {
   const createdMoment = moment.utc(workflow.created);
   const startedMoment = moment.utc(workflow.progress.run_started_at);
   const finishedMoment = moment.utc(workflow.progress.run_finished_at);
+  const stoppedMoment = moment.utc(workflow.progress.run_stopped_at);
+  // Mapping between workflow status and the end moment to use for calculating the duration
+  const endMomentStatusMapping = {
+    finished: finishedMoment,
+    stopped: stoppedMoment,
+    deleted: finishedMoment,
```

Comment:
This does not work if the workflow is first stopped and then deleted, as `run_finished_at` is not set and the duration keeps increasing. What do you think about calculating the duration like this? (not tested)
```js
workflow.duration = formatDuration(
  getDuration(startedMoment, stoppedMoment || finishedMoment)
);
```

Reply:
This would not work because `stoppedMoment` is not evaluated to `false` even if `workflow.progress.run_stopped_at` is `null`, because it still gets converted to a (invalid) `Moment` object. I also liked the mapping of the statuses to the "end moment", especially for readability and extensibility reasons, in case we ever wanted to display something like "queued for X minutes", but thinking about it idk how useful that could be. Anyway, I ended up slightly changing the way in which the end moment for the deleted status is calculated:
```js
deleted: finishedMoment.isValid() ? finishedMoment : stoppedMoment
```
What do you think?